oracle/orchestrator/orchestrator.go

```diff
@@ -91,131 +92,31 @@ func NewProviderOrchestrator(
-	if err != nil {
-		return ProviderState{}, fmt.Errorf("failed to create %s's provider market map: %w", cfg.Name, err)
-	}
-
-	// Select the query handler based on the provider's configuration.
-	var provider *types.PriceProvider
-	switch {
-	case cfg.API.Enabled:
-		queryHandler, err := o.createAPIQueryHandler(cfg, market)
-		if err != nil {
-			return ProviderState{}, fmt.Errorf("failed to create %s's api query handler: %w", cfg.Name, err)
-		}
-
-		provider, err = types.NewPriceProvider(
-			base.WithName[mmtypes.Ticker, *big.Int](cfg.Name),
-			base.WithLogger[mmtypes.Ticker, *big.Int](o.logger),
-			base.WithAPIQueryHandler(queryHandler),
-			base.WithAPIConfig[mmtypes.Ticker, *big.Int](cfg.API),
-			base.WithIDs[mmtypes.Ticker, *big.Int](market.GetTickers()),
-			base.WithMetrics[mmtypes.Ticker, *big.Int](o.providerMetrics),
-		)
-		if err != nil {
-			return ProviderState{}, fmt.Errorf("failed to create %s's provider: %w", cfg.Name, err)
-		}
-	case cfg.WebSocket.Enabled:
-		queryHandler, err := o.createWebSocketQueryHandler(cfg, market)
-		if err != nil {
-			return ProviderState{}, fmt.Errorf("failed to create %s's web socket query handler: %w", cfg.Name, err)
-		}
-
-		provider, err = types.NewPriceProvider(
-			base.WithName[mmtypes.Ticker, *big.Int](cfg.Name),
-			base.WithLogger[mmtypes.Ticker, *big.Int](o.logger),
-			base.WithWebSocketQueryHandler(queryHandler),
-			base.WithWebSocketConfig[mmtypes.Ticker, *big.Int](cfg.WebSocket),
-			base.WithIDs[mmtypes.Ticker, *big.Int](market.GetTickers()),
-			base.WithMetrics[mmtypes.Ticker, *big.Int](o.providerMetrics),
-		)
-		if err != nil {
-			return ProviderState{}, fmt.Errorf("failed to create %s's provider: %w", cfg.Name, err)
-		}
-	default:
-		return ProviderState{}, fmt.Errorf("provider %s has no enabled query handlers", cfg.Name)
-	}
-
-	return ProviderState{
-		Provider: provider,
-		Market:   market,
-		Enabled:  len(market.GetTickers()) > 0,
-		Cfg:      cfg,
-	}, nil
-}
-
-// GetProviderState returns all of the providers and their state.
-func (o *ProviderOrchestrator) GetProviderState() map[string]ProviderState {
+// GetProviders returns all of the providers.
+func (o *ProviderOrchestrator) GetPriceProviders() []types.PriceProviderI {
 	o.mut.Lock()
 	defer o.mut.Unlock()
 
-	// Copy the providers
-	providers := make(map[string]ProviderState, len(o.providers))
-	maps.Copy(providers, o.providers)
-
-	return providers
-}
-
-// createAPIQueryHandler creates a new API query handler for the given provider configuration.
-func (o *ProviderOrchestrator) createAPIQueryHandler(
-	cfg config.ProviderConfig,
-	market types.ProviderMarketMap,
-) (types.PriceAPIQueryHandler, error) {
-	if o.apiQueryHandlerFactory == nil {
-		return nil, fmt.Errorf("cannot create provider; api query handler factory is not set")
+	providers := make([]types.PriceProviderI, 0, len(o.providers))
+	for _, state := range o.providers {
+		providers = append(providers, state.Provider)
```

Comment:
```suggestion
	providers := make([]types.PriceProviderI, len(o.providers))
	for i, state := range o.providers {
		providers[i] = state.Provider
```

Reply:
this is iterating over a map not a list :(